common/src/main/kotlin/streams/extensions/CommonExtensions.kt

```diff
@@ -13,6 +14,17 @@ fun String.toPointCase(): String {
     return this.split("(?<=[a-z])(?=[A-Z])".toRegex()).joinToString(separator = ".").toLowerCase()
 }
 
-fun String.quote(): String {
-    return if (SourceVersion.isIdentifier(this)) this else "`$this`"
+fun String.quote(): String = if (SourceVersion.isIdentifier(this)) this else "`$this`"
+
+fun Map<String, Any?>.flatten(map: Map<String, Any?> = this, prefix: String = ""): Map<String, Any?> {
+    return map.flatMap {
+        val key = it.key
+        val value = it.value
+        val newKey = if (prefix != "") "$prefix.$key" else key
+        if (value is Map<*, *>) {
+            flatten(value as Map<String, Any>, newKey).entries
+        } else {
+            listOf(AbstractMap.SimpleEntry(newKey, value))
```

Comment:
why this? and not a tuple?

Reply:
can also fix these two later

Reply:
Ok

Reply:
I mean I fixed that, I was not aware of the `toList` method that transforms the entries into `List<Pair<K,V>>` thanks